### PR TITLE
Feat(refs:T31913): Hide press announcement if no press news existing

### PR DESCRIPTION
**Pull request by elmasdemos**
**Ticket:** https://yaits.demos-deutschland.de/T31913

### Description: 

- press news section shall be hidden for all projects, if no press news existing
- in this the width of the global news get added by the width of the press announcement section

### How to review/test

- go to startpage (you dont need to login)
- go to the section `Aktuelle Meldungen` and click to `Alle Meldungen`
- if press news not existing you will only see the global news
- if you have no press news activated, login as `Mandanten-Administration`and click to `Portal-Mitteilungen`
  -> here you can activate press news to show it in `Alle Meldungen` or create a new one
  -> than logout and go again to the startpage to `Alle Meldungen`

### Linked PRs (optional)

- https://github.com/demos-europe/demosplan-project-diplanbau/pull/80
- https://github.com/demos-europe/demosplan-project-blp/pull/38

### PR Checklist

- [x] Link all relevant tickets
- [x] Move the tickets on the board accordingly

### CODE REVIEW

File: templates/bundles/DemosPlanCoreBundle/DemosPlanNews/globalnewslist_list_index.html.twig [hunk 1]
```diff
@@ -6,8 +6,10 @@
     }%}
 
         {% block content %}
-
-            <div class="layout__item u-1-of-1-palm{% if hidePressAnnouncements is not defined %} u-1-of-2-lap-up{% endif %} u-pl-0_5 u-pl-0-lap-up u-pr-lap-up">
+            <div class="layout__item u-1-of-1-palm  u-pl-0_5 u-pl-0-lap-up u-pr-lap-up
+                {% if templateVars.pressList is not empty %}
```
Comment (spiess-demos) [inline templates/bundles/DemosPlanCoreBundle/DemosPlanNews/globalnewslist_list_index.html.twig:10]: `hidePressAnnouncements` is still used within `globalnewslist_list_index.html.twig`. To keep the current behavior we could chain the two conditions like so:

```suggestion
                {% if templateVars.pressList is not empty or hidePressAnnouncements is not defined %}
```

**Review by spiess-demos — CHANGES_REQUESTED** [review]

File: templates/bundles/DemosPlanCoreBundle/DemosPlanNews/globalnewslist_list_index.html.twig (hunk 1 above)
Comment (elmasdemos) [reply templates/bundles/DemosPlanCoreBundle/DemosPlanNews/globalnewslist_list_index.html.twig:10]: its not needed this way. look at PR https://github.com/demos-europe/demosplan-project-blp/pull/38

File: templates/bundles/DemosPlanCoreBundle/DemosPlanNews/globalnewslist_list_index.html.twig [hunk 2]
```diff
@@ -18,7 +20,7 @@
                 {% endif %}
             </div>
 
-            {% if hidePressAnnouncements is not defined %}
+            {% if templateVars.pressList is not empty %}
```
[the remark this answers is not in the document]
Comment (elmasdemos) [reply templates/bundles/DemosPlanCoreBundle/DemosPlanNews/globalnewslist_list_index.html.twig:23]: in the project repo it has no effect: the logic was incomplete and the folder structure was not correct

**Comment by spiess-demos** [comment]
With these changes and the ones within https://github.com/demos-europe/demosplan-project-blp/pull/38, the press section is shown as soon as one news has the "press" category. If this is intended, i'd be fine with the solution.

**Pull request merged**